src/rust/engine/process_execution/src/named_caches.rs

```diff
@@ -51,13 +51,27 @@ impl NamedCaches {
   pub fn local_paths<'a>(
     &'a self,
     caches: &'a BTreeMap<CacheName, RelativePath>,
-  ) -> impl Iterator<Item = WorkdirSymlink> + 'a {
-    caches
+  ) -> Result<Vec<WorkdirSymlink>, String> {
+    let symlinks = caches
       .iter()
       .map(move |(cache_name, workdir_rel_path)| WorkdirSymlink {
         src: workdir_rel_path.clone(),
         dst: self.local_base.join(&cache_name.0),
       })
+      .collect::<Vec<_>>();
+
+    for symlink in &symlinks {
+      safe_create_dir_all_ioerror(&symlink.dst).map_err(|err| {
+        format!(
+          "Error making symlink {} -> {}: {:?}",
+          symlink.src.display(),
+          symlink.dst.display(),
+          err
+        )
+      })?
```

Comment:
This operation doesn't actually create the symlink: there is a call to `symlink` near the source of this code that does that which probably needs to move as well.

Reply:
I'll fix the error message. The symlink creation needs to stay where it is because, with the Docker command runner, it deals with the destination modified to account for the container-specific path. This function has no knowledge of what the container base path for the symlink will be.

Reply:
And PR title too then probably.